### ISSUE TITLE
Difference between optional and required Records

### ISSUE THREAD

**Issue by Marcisbee** [ccontrols/structured-types#7]
There's a difference in exported data between optional and required Records.

# Description

Given this code:
```tsx
export interface MyProps {
  first: Record<string, string>;
  second?: Record<string, string>;
}

export default function MyComponent({
  first = {},
  second = {},
}: MyProps) {
  return <div />;
}
```

structured-types exports data like this:
```json
{
  "default": {
    "name": "MyComponent",
    "extension": "react-prop-types",
    "kind": 11,
    "properties": [
      {
        "parent": "MyProps",
        "name": "first",
        "kind": 15,
        "properties": []
      },
      {
        "parent": "MyProps",
        "optional": true,
        "name": "second",
        "kind": 15,
        "properties": [],
        "filePath": "/lib.es5.d.ts",
        "generics": [
          {
            "name": "K"
          },
          {
            "name": "T"
          }
        ],
        "description": "Construct a type with a set of properties K of type T",
        "type": "Record"
      }
    ]
  },
  "MyProps": {
    "name": "MyProps",
    "kind": 14,
    "properties": [
      {
        "name": "first",
        "kind": 15,
        "generics": [
          {
            "name": "K"
          },
          {
            "name": "T"
          }
        ]
      },
      {
        "optional": true,
        "name": "second",
        "kind": 15,
        "type": "Record",
        "generics": [
          {
            "kind": 1
          },
          {
            "kind": 1
          }
        ]
      }
    ]
  }
}
```

It's clear that `first` is drastically different from `second` in terms of returned data.

Is this a bug or am I missing some configuration?

# Reproduction

Here's a link to playground: https://structured-types.vercel.app/?code=%22export%20interface%20MyProps%20%7B%5Cn%20%20first%3A%20Record%3Cstring%2C%20string%3E%3B%5Cn%20%20second%3F%3A%20Record%3Cstring%2C%20string%3E%3B%5Cn%7D%5Cn%5Cnexport%20default%20function%20MyComponent%28%7B%5Cn%20%20first%20%3D%20%7B%7D%2C%5Cn%20%20second%20%3D%20%7B%7D%2C%5Cn%7D%3A%20MyProps%29%20%7B%5Cn%20%20return%20%3Cdiv%20%2F%3E%3B%5Cn%7D%22&parseOptions=%7B%22internalTypes%22%3A%5B%22Function%5CnCallableFunction%5CnNewableFunction%5CnString%5CnBoolean%5CnBooleanish%5CnNumber%5CnArray%5CnConcatArray%5CnReadonlyArray%5CnTemplateStringsArray%22%5D%2C%22extensions%22%3A%5B%22react%22%2C%22react-prop-types%22%5D%7D

**Comment by atanasster** [comment]
@Marcisbee - seems like a bug, for some reason the typescript compiler is returning different types, solely depending on whether the prop is required or not.

I will investigate further and let you know.

**Comment by atanasster** [comment]
@Marcisbee , just released 3.21.0 and it should be fixed

Here is your example in the playground: https://bit.ly/38NSEJJ

Also your example is now part of the test suite, so this issue will hopefully not re-appear.

Many thanks for submitting a detailed issue, please create a new issue if needed
